agents-common/src/main/java/org/apache/ranger/plugin/service/RangerBasePlugin.java

```diff
@@ -883,14 +883,11 @@ public RangerRoles getRangerRoles() {
     }
 
     public Set<RangerRole> getRangerRoleForPrincipal(String principal, String type) {
-        Set<RangerRole>          ret         = new HashSet<>();
-        Set<RangerRole>          rangerRoles = null;
+        Set<RangerRole>          ret          = new HashSet<>();
+        RangerPolicyEngine       policyEngine = this.policyEngine;
```

Comment:
Do we need this change ?. I can see  getRangerRole is already checking policyEngine  for null as follows
 RangerPolicyEngine policyEngine = this.policyEngine;

        if (policyEngine != null) {
            return policyEngine.getRangerRoles();
        }

Reply:
@vyommani  - which line number are you referring to?  Local variable`policyEngine` (line 887) is referenced later in line 893. Hence it is necessary to check for null value here.

Reply:
Yes, you're right. Upon reviewing the code changes, I realized I was only focusing on the modified code.